File: Problems/No.1337_The_K_Weakest_Rows_in_a_Matrix/the_k_weakest_rows_in_a_matrix.py
class Solution(object):
    def kWeakestRows(self, mat, k):
        """
        :type mat: List[List[int]]
        :type k: int
        :rtype: List[int]
        """

        # parameters
        Row_Strength = []
        kWeakestRows = []

        # start computing
        # calculate soldiers in each row & order them
        for i in range(len(mat)):
            soldiers = 0
            for s in mat[i]:
                if s:
                    soldiers += 1
                else:
                    break

            heapq.heappush(Row_Strength, [soldiers, i])

        # k weakest rows
        for i in range(k):
            kWeakestRows.append(heapq.heappop(Row_Strength)[1])

        return kWeakestRows

File: Problems/No.1337_The_K_Weakest_Rows_in_a_Matrix/the_k_weakest_rows_in_a_matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        
        // parameters
        std::priority_queue<std::pair<int, int>> Row_Strength;
        std::vector<int> kWeakestRows;

        // start computing
        // calculate soldiers in each row & order them
        for (int i=0; i<mat.size(); i++) {

            auto soldiers = 0;
            for (int s=0; s<mat[i].size(); s++) {

                if (mat[i][s] == 1) soldiers++;
                else break;

            }

            Row_Strength.push({soldiers, i});
            if (i >= k) Row_Strength.pop();

        }

        // k weakest rows
        while (!Row_Strength.empty()) {

            kWeakestRows.insert(kWeakestRows.begin(), Row_Strength.top().second);
            Row_Strength.pop();

        }

        return kWeakestRows;

    }
};
